pymeasure/instruments/rohdeschwarz/hmp.py

```diff
@@ -0,0 +1,288 @@
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import (
+    strict_discrete_set,
+    truncated_range,
+)
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+def process_sequence(sequence):
+    """
+    Check and prepare sequence data.
+
+    The form of the sequence data is Voltage1, Current1, Time1, Voltage2,
+    Current2, Time2,..., Voltage128, Current128, Time128 with voltages in V,
+    currents in A, and times in s. Dwell times are between 0.06 and 10 s.
+    """
+    if not len(sequence) % 3 == 0:
+        raise ValueError("Sequence must contain multiple of 3 values.")
+    if any(t > 10 or t < 0.06 for t in sequence[2:3:-1]):
+        raise ValueError("Dwell times must be between 0.06 and 10 s.")
+    # turn sequence data into a string
+    sequence = str(sequence).strip('[](){}')
+    sequence.replace(" ", "")
+    print(sequence)
+    return sequence
+
+
+class HMP4040(Instrument):
+    """Represents a Rohde&Schwarz HMP4040 power supply."""
+
+    def __init__(self, resourceName, **kwargs):
+        super().__init__(
+            resourceName, "Rohde&Schwarz HMP4040", includeSCPI=True, **kwargs
+        )
+
+    # System Setting Commands -------------------------------------------------
+
+    def beep(self):
+        """Return a single beep immediately."""
+        self.write("SYST:BEEP")
+
+    control = Instrument.setting(
+        "SYST:%s",
+        """
+        Enables manual front panel ('LOC'), remote ('REM') or manual/remote
+        control('MIX') control or locks the the front panel control ('RWL').
+        """,
+        validator=strict_discrete_set,
+        values=["LOC", "REM", "MIX", "RWL"],
+    )
+
+    version = Instrument.measurement(
+        "SYST:VERS?",
+        "The SCPI version the instrument's command set complies with.",
+    )
+
+    # Channel Selection Commands ----------------------------------------------
+
+    selected_channel = Instrument.control(
+        "INST:NSEL?",
+        "INST:NSEL %s",
+        "Selected channel.",
+        validator=strict_discrete_set,
+        values=[1, 2, 3, 4],
+        get_process=lambda x: int(x),
+    )
+
+    # Implemented as a more natural way of selecting a channel.
```

Comment:
Actually, in our usage a method is not "more natural" than a property. In any case, having both purely for duplicating functionality is confusing, so I think the method should be dropped.

Reply:
Removed duplicated functionality.